sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/endpoints.py

```diff
@@ -142,8 +142,9 @@ class ApacheFilters(object):
         'default_outcome',
         'outcomes',
         'capabilities'
-    ])
-Source.__new__.__defaults__ = (None,) * len(Source._fields) # type: ignore
+    ],
+    defaults=(None,) * 11 # type: ignore
```

Comment:
Just curious, what is this change doing?

Reply:
Its using the new way from py3.7+ onwards for setting a namedtuple fields default values, an iterable of None will make all of them None https://docs.python.org/3/library/collections.html#collections.namedtuple